usecases/decision_phantom_usecase.go

```diff
@@ -51,42 +51,27 @@ func (usecase *DecisionPhantomUsecase) CreatePhantomDecision(ctx context.Context
 		return models.DecisionWithRuleExecutions{},
 			fmt.Errorf("error evaluating scenario: %w", err)
 	}
-	if testRunScenarioExecution.ScenarioId != "" {
-		decision := models.AdaptScenarExecToDecision(testRunScenarioExecution, input.Payload, nil)
-		if !WithRuleExecutionDetails {
-			for i := range decision.RuleExecutions {
-				decision.RuleExecutions[i].Evaluation = nil
-			}
-		}
-		ctx, span = tracer.Start(
-			ctx,
-			"DecisionUsecase.CreateDecision.store_phantom_decision",
-			trace.WithAttributes(attribute.String("scenario_id", input.Scenario.Id)),
-			trace.WithAttributes(attribute.Int("nb_rule_executions", len(decision.RuleExecutions))))
-		defer span.End()
-
-		newDecision, err := executor_factory.TransactionReturnValue(ctx, usecase.transactionFactory, func(
-			tx repositories.Transaction,
-		) (models.DecisionWithRuleExecutions, error) {
-			if err = usecase.repository.StorePhantomDecision(
-				ctx,
-				tx,
-				decision,
-				input.OrganizationId,
-				testRunScenarioExecution.ScenarioIterationId,
-				decision.DecisionId,
-			); err != nil {
-				return models.DecisionWithRuleExecutions{},
-					fmt.Errorf("error storing decision: %w", err)
-			}
+	if testRunScenarioExecution.ScenarioId == "" {
+		return models.DecisionWithRuleExecutions{}, err
+	}
+	decision := models.AdaptScenarExecToDecision(testRunScenarioExecution, input.ClientObject, nil)
```

Comment:
one last thing here: we discussed having a separate model for phantom decisions (hence also a separate adapter here), I still think that would be a good idea :-)